CMSgov/design-system
# [WNMGDS-602] Change close button prop name in Dialog component

**Note:** 
- Add the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) to the PR title like this `[WNMGDS-10] - title of pr here` to link to the related issue in Jira.
- You can automatically [close related GitHub issues by using keywords](https://help.github.com/en/articles/closing-issues-using-keywords).
- A demo doc site is automatically generated for this branch at `http://design-system-demo.s3-website-us-east-1.amazonaws.com/{branch_name}`. For example, the demo for `WNMGDS-10/feature` will be located at `http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-10/feature`.
- If your changes involve code please update the snapshots by running `yarn update-snapshots`.

## Summary
Both Dialog and HelpDrawer have props for the close button, but they are named different. We should make `closeButtonText` and `closeText` naming and usage consistent.

Bernard also had this note in the ticket `We also need to consider usage in MCT where an icon is used instead of text.`

**But since we are just renaming, and not changing the type (which is ReactNode), I assume that if it supported an icon before, it still will. Is there something I'm missin ghere?**

### Changed
Changed `closeText` to `closeButtonText` in Dialog, to be consistent with the HelpDrawer prop name.

### Deprecated
`closeText` is being deprecated. Use `closeButtonText` instead

## How to test
Demo site: http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-602/change-closeButton-prop-name

- Make sure on Demo site everything still looks correct, and closeText is gone and closeButtonText shows
- Run branch locally
- In the `Dialog.example.jsx` file, send in `closeText` as a prop. You should see a warning like below in the console, but it should still work.
![image](https://user-images.githubusercontent.com/21293083/112994240-07ccf200-9138-11eb-9dc4-f177ad15ae15.png)

- Send in `closeButtonText` and it should work without warning.
- Send in nothing and it should default to Close without a warning.
